### PR TITLE
favbar

### PR DESCRIPTION
**Pull request by sbcgua**
To #401 

image

### CODE REVIEW

**Comment by sbcgua** [comment]
Ah no, hold a bit with this PR pls, I forgot a "new/clones repo to favs" feature

File: src/zabapgit_html.prog.abap [hunk 1]
```diff
@@ -250,15 +253,17 @@ CLASS lcl_html_toolbar IMPLEMENTATION.
...
     ls_item-opt = iv_opt.
     ls_item-typ = iv_typ.
     APPEND ls_item TO mt_items.
   ENDMETHOD.  "add
 
-  METHOD render.
+  METHOD render. "TODO refactor
 
     DATA: lv_class   TYPE string,
+          lv_text    TYPE string,
```
Comment (EduardoCopat) [inline src/zabapgit_html.prog.abap]: Varaible not used?

**Comment by sbcgua** [comment]
added toggle_favorite after clone & newoffline.
removed lv_text, indeed forgot it, thanks. favbar is fine - this is naming for all menus more or less. 

@larshp there is a bug with locks but I don't know locks well so not sure how to fix. 
Actually toggle_fav after clone or new does not work. It raises error `Could not aquire lock`. It does not happen in debugger, seems like the code lock-release-lock too quickly and 'release' does not complete. Could you have a look ?

e.g. `ZABAPGIT_SERVICES_REPO->new_offline`
```
    lcl_app=>user( )->set_repo_show( lo_repo->get_key( ) ). " <<< LOCK 1
    toggle_favorite( lo_repo->get_key( ) ).                  " <<< LOCK 2
```

**Comment by larshp** [comment]
locking: try adding COMMIT WORK AND WAIT between the 2 statements (I think we have 2 open issues on looking into locking)

**Comment by sbcgua** [comment]
didn't work. Can it be because `set_repo_show` issues `COMMIT WORK` without `WAIT` just before ?

**Comment by larshp** [comment]
yes, adding "AND WAIT" in set_repo_show will fix the issue

**Comment by larshp** [comment]
can we extend the icon element into the red area? I typically move my mouse in a downward left direction with dropdowns, and when I move the mouse into the red area, the dropdown disappears. It looks like there is enough space to extend the element,

image

**Comment by sbcgua** [comment]
moving `wait` to `set_repo_show ` fixed the issue, thanks (not commited yet).

icon: so the question is about mouse movement, not about icon size, right ? I think I'll fix it another way, design will remain the same. It might not work for IE7 ... but I guess it is OK. Have a look in the evening.

**Comment by larshp** [comment]
when the WAIT/locking issue is fixed I guess we can merge? The mouse movement is a minor issue which we can always fix in a follow up commit

**Comment by sbcgua** [comment]
OK, agree. Then can be merged. I amended the previous commit so it includes `wait`.

**Pull request merged**